tests/kernel/timer/timer_monotonic/src/main.c

```diff
@@ -8,21 +8,132 @@
+int count = 0;
+
+
+
+static void callback_period(struct k_timer *timer)
+{
+	u32_t pct;
+
+	stop_time = k_cycle_get_32();
+	cycles_spent = stop_time - start_time;
+	nanoseconds_spent = (u32_t)k_cyc_to_ns_floor64(cycles_spent);
+	pct = (u64_t)cycles_spent * 100000U / sys_clock_hw_cycles_per_sec();
+	start_time = stop_time;
+	pctlist[count] = pct;
+	time_spent[count] = nanoseconds_spent;
+	cycles_cost[count] = cycles_spent;
+
+	pct =0;
+
+	count++;
+	if (count >= 10) {
+		k_timer_stop(timer);
+	}
+}
+
+static void callback_duration(struct k_timer *timer)
 {
-	u32_t start, end, delta, pct;
+	stop_time = k_cycle_get_32();
+}
+
+int test_periodtimer_frequency(void)
+{
+	int i;
+
+	TC_PRINT("Testing system tick frequency\n");
+
+	k_sleep(1);
+	k_timer_init(&test_periodtimer, callback_period, NULL);
+	start_time = k_cycle_get_32();
+	k_timer_start(&test_periodtimer, 1, 1);
+	k_sleep(1000);
+	k_timer_stop(&test_periodtimer);
+	for(i = 0; i < 10; i++)
+	{
+		printk("delta: %u expected cycles: %u "
+				"expected time(nanosec): %u "
+				"cost time(nanosec): %u  %u%%\n",
+				cycles_cost[i], sys_clock_hw_cycles_per_sec()/1000,
+				1000000 ,time_spent[i], pctlist[i]);
+		/* Heuristic: if we're more than 10% off, throw an error */
+		if (pctlist[i] < 90 || pctlist[i] > 110) {
+			TC_PRINT("Clock calibration is way off!\n");
+			return -1;
+			break;
+		}
+		time_spent[i] = 0;
+		cycles_cost[i] = 0;
+		pctlist[i] = 0;
+	}
+
+	start_time = 0;
+	stop_time = 0;
+	cycles_spent = 0;
+	nanoseconds_spent = 0;
+
+	return 0;
+}
+
+int test_oneshottimer_frequency(void)
+{
+	u32_t delta, pct;
+
+	TC_PRINT("Testing system tick frequency\n");
+
+	k_sleep(1);
+	k_timer_init(&test_timer, callback_duration, NULL);
+	start_time = k_cycle_get_32();
+
+	k_timer_start(&test_timer, 1, 0);
+	k_timer_status_sync(&test_timer);
+	delta = stop_time - start_time;
+	nanoseconds_spent = (u32_t)k_cyc_to_ns_floor64(delta);
+	k_timer_stop(&test_timer);
+	pct = (u64_t)delta * 100000U / sys_clock_hw_cycles_per_sec();
+
+	printk("delta: %u expected cycles: %u "
+			"expected time(nanosec): %u "
+			"cost time(nanosec): %u  %u%%\n",
+			delta, sys_clock_hw_cycles_per_sec()/1000,
+			1000000, nanoseconds_spent, pct);
+
+	start_time = 0;
+	stop_time = 0;
+	nanoseconds_spent = 0;
+	/* Heuristic: if we're more than 10% off, throw an error */
+	if (pct < 90 || pct > 110) {
+		TC_PRINT("Clock calibration is way off!\n");
+		return -1;
+	}
+
+	return 0;
+}
+
+int test_frequency_second(void)
+{
+	u32_t start, end, delta, nanoseconds_spent, pct;
 
 	TC_PRINT("Testing system tick frequency\n");
 
 	start = k_cycle_get_32();
 	k_sleep(K_MSEC(1000));
 	end = k_cycle_get_32();
-
 	delta = end - start;
+	nanoseconds_spent = (u32_t)k_cyc_to_ns_floor64(delta);
 	pct = (u64_t)delta * 100U / sys_clock_hw_cycles_per_sec();
 
-	printk("delta: %u  expected: %u  %u%%\n", delta,
-	       sys_clock_hw_cycles_per_sec(), pct);
+	printk("delta: %u expected cycles: %u "
```

Comment:
I know it was in the original code, but TC_PRINT is preferred over a bare printk() in test code now that the log subsystem is in regular use.